core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -424,9 +408,87 @@ public boolean delete(PageId pageId) {
     }
   }
 
+  @Override
+  public State state() {
+    return mState.get();
+  }
+
+  /**
+   * Restores a page store a the configured location, updating meta store.
+   */
+  private void restoreOrInit(PageStoreOptions options) {
+    Preconditions.checkState(mState.get() == READ_ONLY);
+    if (!restore(options)) {
+      try {
+        mPageStore.close();
+        try (LockResource r = new LockResource(mMetaLock.writeLock())) {
+          mMetaStore.reset();
+        }
+        PageStore.initialize(options);
+        mPageStore = PageStore.create(options);
+      } catch (Exception e) {
+        LOG.error("Failed to clean up PageStore", e);
+        return;
+      }
+    }
+    mState.set(READ_WRITE);
+    Metrics.STATE.inc();
+  }
+
+  private boolean restore(PageStoreOptions options) {
+    LOG.info("Attempt to restore PageStore with {}", options);
+    Path rootDir = Paths.get(options.getRootDir());
+    if (!Files.exists(rootDir)) {
+      LOG.error("Failed to restore PageStore: Directory {} does not exist", rootDir);
+      return false;
+    }
+    long discardedPages = 0;
+    long discardedBytes = 0;
+    try (Stream<PageInfo> stream = mPageStore.getPages()) {
+      Iterator<PageInfo> iterator = stream.iterator();
+      while (iterator.hasNext()) {
+        PageInfo pageInfo = iterator.next();
+        if (pageInfo == null) {
+          LOG.error("Invalid page info");
+          return false;
+        }
+        PageId pageId = pageInfo.getPageId();
+        ReadWriteLock pageLock = getPageLock(pageId);
+        try (LockResource r = new LockResource(pageLock.readLock())) {
```

Comment:
Should this be a write lock because we might delete the page if there is not enough space?

Reply:
good catch. corrected